### PR TITLE
Feature avx2 mxfp4i8 gemm

### PR DESCRIPTION
**Pull request by egeor**
This PR add support for mxfp4i8(bf16/f32) gemm kernel (SRF)

### CODE REVIEW

File: src/generator_gemm_avx2_microkernel.c [hunk 1]
```diff
@@ -124,7 +145,7 @@ void libxsmm_generator_gemm_avx2_microkernel_Amxfp4( libxsmm_generated_code*
         l_tmp_reg_acc_start + l_i, l_tmp_reg_acc_start + l_i, l_tmp_reg_acc_start + l_i );
   }
 
-  if (16/l_k_unroll_factor > 1) {
+  if ((32/l_k_pack_factor)/l_k_unroll_factor > 1) {
```
Comment (alheinecke) [inline src/generator_gemm_avx2_microkernel.c:148]: I assume, as this is not guarded by an l_is_Amxfp4_Bi8_gemm, it's valid for all precision but changes unrolling behavior.

File: src/generator_gemm_avx2_microkernel.c [hunk 2]
```diff
@@ -280,45 +358,120 @@ void libxsmm_generator_gemm_avx2_microkernel_Amxfp4( libxsmm_generated_code*
...
+                                                    (l_k % 2 == 0) ? l_m_vreg_k0 : l_m_vreg_k1,
+                                                    l_tmp_reg_acc_start + l_m + (l_m_blocking * l_n) );
+        }
+      }
+    }
   }
 
   /* Increase B for next K iteration */
-  if (16/l_k_unroll_factor > 1 || i_k > 32) {
+  if ((32/l_k_pack_factor)/l_k_unroll_factor > 1 || i_k > 32) {
```
Comment (alheinecke) [inline src/generator_gemm_avx2_microkernel.c:389]: same as above

File: src/generator_gemm_avx2_microkernel.c [hunk 3]
```diff
@@ -328,11 +481,17 @@ void libxsmm_generator_gemm_avx2_microkernel_Amxfp4( libxsmm_generated_code*
...
+                                   i_micro_kernel_config->alu_add_instruction,
+                                   i_gp_reg_mapping->gp_reg_zpt,
+                                   (long long)4*i_micro_kernel_config->datatype_size_in2 );
+    }
     /* End of K loop */
     libxsmm_generator_generic_loop_footer_with_idx_inc( io_generated_code, i_micro_kernel_config->io_loop_label_tracker, i_gp_reg_mapping->gp_reg_kloop, 32, i_k);
   }
   /* Adjust reg_b */
-  if (16/l_k_unroll_factor > 1 || i_k > 32) {
+  if ((32/l_k_pack_factor)/l_k_unroll_factor > 1 || i_k > 32) {
```
Comment (alheinecke) [inline src/generator_gemm_avx2_microkernel.c:494]: same as above.

**Review by alheinecke — APPROVED** [review]
Some minor comment/question.

File: src/generator_gemm_avx2_microkernel.c (hunk 1 above)
Comment (egeor) [reply src/generator_gemm_avx2_microkernel.c:148]: Correct, thus generalized this 16 to 32/l_k_pack_factor (which is capturing the i8/bf16 logic for mxfp4)

File: src/generator_gemm_avx2_microkernel.c (hunk 2 above)
Comment (egeor) [reply src/generator_gemm_avx2_microkernel.c:389]: see above

File: src/generator_gemm_avx2_microkernel.c (hunk 3 above)
Comment (egeor) [reply src/generator_gemm_avx2_microkernel.c:494]: see above

**Pull request merged**